adoc/extensions/sycl_khr_queue_empty_query.adoc

```diff
@@ -0,0 +1,86 @@
+[[sec:khr-queue-empty-query]]
+= sycl_khr_queue_empty_query
+
+This extension allows developers to query the queue's emptiness, meaning if all
+commands submitted to a queue have been completed.
+
+[[sec:khr-queue-empty-query-dependencies]]
+== Dependencies
+
+This extension has no dependencies on other extensions.
+
+[[sec:khr-queue-empty-query-feature-test]]
+== Feature test macro
+
+An implementation supporting this extension must predefine the macro
+[code]#SYCL_KHR_QUEUE_EMPTY_QUERY# to one of the values defined in the table
+below.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|Initial version of this extension.
+|===
+
+
+[[sec:khr-queue-empty-query-funct]]
+== New Queue Function to Query Emptiness
+
+This extension adds the following function to the [code]#sycl::queue# class,
+which provides information about the emptiness of the queue.
+
+'''
+
+.[apidef]#queue::khr_empty#
+[source,role=synopsis,id=api:queue-khr-empty]
+----
+bool khr_empty() const
+----
```

Comment:
I forget if we already discussed this or not, but should this be called `is_empty()`?

Somebody might call `queue.empty()` expecting it to _empty_ the queue of its contents.  It's more obvious that `queue.is_empty()` tests a condition.

I know that there is an `empty()` in `std::vector` and a `std::empty()` free-function, but those are defined for containers.  `queue` is _not_ a container (even if it has the same name as one), so it may be in our interest to use names that are very clearly different to those in the container interface.

I'd be interested to hear what @nliber thinks about this.

Reply:
Good point, maybe`is_empty()` is a better name to avoid confusion. I will bow to Nevin recommendation :) 

(And to lament, it's sad ct that we cannot use `?` in  c++. In ruby, methods that return a bool have their name ended with a `?`, like `empty?`. Beautiful)

Reply:
Tried with `c3330dc1994e`

Reply:
WG discussed, and had concencous on `empty`